### PR TITLE
vhost-device-sound: implement RX capture with ALSA

### PR DESCRIPTION
**Pull request by epilys**
Re-created #476

### Summary of the PR

> Add an RX code path that is similar to the existing TX code in the ALSA backend.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Review by vireshk — APPROVED** [review]

**Review by dorindabassey — APPROVED** [review]

**Review by MatiasVara — APPROVED** [review]

**Review by stsquad — APPROVED** [review]

File: staging/vhost-device-sound/src/device.rs [hunk 1]
```diff
@@ -681,7 +681,7 @@ impl VhostUserSoundBackend {
             threads,
             virtio_cfg: VirtioSoundConfig {
                 jacks: 0.into(),
-                streams: 1.into(),
+                streams: Le32::from(streams_no as u32),
```
Comment (stefano-garzarella) [inline staging/vhost-device-sound/src/device.rs:684]: This is a pre-existing issue, right?
I mean, it's true that we didn't support RX yet, but we already had 2 streams allocated.

Maybe better to put in another patch (not a strong opinion)

**Review by stefano-garzarella — COMMENTED** [review]
Can we split the device and ALSA support for RX in two patches?

I'd also suggest writing in the commit what is missing (e.g. tests, pipewire support, etc.)

Anyway, the code LGTM! Thanks!

File: staging/vhost-device-sound/src/device.rs (hunk 1 above)
Comment (epilys) [reply staging/vhost-device-sound/src/device.rs:684]: The issue was that returning `Le32::from(streams_no as u32)` would show a capture stream to the guest but it wouldn't work at all. Now it will work depending on backend support. IMO we should change the VirtioSoundConfig value to runtime computed values depending on the audio backend, since it has to query the host for what streams are available, what channel maps are available, etc. We should do that in a separate PR.

**Comment by epilys** [comment]
> Can we split the device and ALSA support for RX in two patches?

Hm, that'd have been better indeed. Oh welp! :D

File: staging/vhost-device-sound/src/device.rs (hunk 1 above)
Comment (stefano-garzarella) [reply staging/vhost-device-sound/src/device.rs:684]: Ack, I agree!
